src/chainstate/stacks/index/trie_sql.rs

```diff
@@ -93,6 +114,63 @@ pub fn create_tables_if_needed(conn: &mut Connection) -> Result<(), Error> {
     tx.commit().map_err(|e| e.into())
 }
 
+fn get_schema_version(conn: &Connection) -> u64 {
+    // if the table doesn't exist, then the version is 1.
+    let sql = "SELECT version FROM schema_version";
+    match conn.query_row(sql, NO_PARAMS, |row| row.get::<_, i64>("version")) {
+        Ok(x) => x as u64,
+        Err(e) => {
+            debug!("Failed to get schema version: {:?}", &e);
+            1u64
+        }
+    }
+}
+
+/// Migrate the MARF database to the currently-supported schema.
+/// Returns the version of the DB prior to the migration.
+pub fn migrate_tables_if_needed<T: MarfTrieId>(
```

Comment:
how long does this take?

Reply:
Just over 15 minutes for migrating all MARFs.